### PR TITLE
Refactor/tidy server

### PR DESCRIPTION
**Pull request by alectrocute**
Hi Lemdevs!

I propose we:

- Refactor our `server/index.tsx` to use a common Express.js pattern (`handlers`, `middleware` and `utils`)
- Add Express types to handler/middleware export parameters
- Make more human readable wherever possible

Screenshot 2023-06-16 at 10 47 26 AM

In future:

- Rewrite/split out `CatchAllHandler` to be simpler if possible

Thanks so much!

### CODE REVIEW

File: src/server/utils/set-forwarded-headers.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    out.host = headers.host;
+  }
+
+  const realIp = headers["x-real-ip"];
+
+  if (realIp) {
+    out["x-real-ip"] = realIp as string;
+  }
+
+  const forwardedFor = headers["x-forwarded-for"];
```
Comment (alectrocute) [inline src/server/utils/set-forwarded-headers.ts:18]: Possible bug here. `headers["x-forwarded-for"]` can be either `string` or `string[]`.

**Review by dessalines — APPROVED** [review]
Thx for doing this, looks much cleaner.

@SleeplessOne1917 and I are getting fairly antsy to start *yet another* lemmy-ui in leptos-rs, with tailwind + DaisyUI , and that will avoid this entire inferno isomorphic mess, since it already has first-class isomorphic support. But that's for a later time.

**Comment by alectrocute** [comment]
> @SleeplessOne1917 and I are getting fairly antsy to start yet another lemmy-ui in leptos-rs, with tailwind + DaisyUI , and that will avoid this entire inferno isomorphic mess, since it already has first-class isomorphic support. But that's for a later time.

I can't wait to get involved with that. That's an awesome stack.

**Comment by dessalines** [comment]
I would 100% support people charging ahead on that without me. Right now I'm just a bit overwhelmed (800+ of the *important* notifications, many more unimportant), and am only responding to PRs and coding.

**Comment by SleeplessOne1917** [comment]
I'll take a closer look at this later, but be aware that the changes in #1043 will likely affect this.

**Comment by alectrocute** [comment]
> I'll take a closer look at this later, but be aware that the changes in #1043 will likely affect this.

That's totally fine, it'll be easy for me to merge the conflicts. Thanks for the heads up!

**Comment by SleeplessOne1917** [comment]
@dessalines Daisy UI may not even be necessary. We can define our own reusable classes using `@apply`. https://tailwindcss.com/docs/functions-and-directives#apply

**Comment by SleeplessOne1917** [comment]
@alectrocute #1043 is merged now.

**Comment by alectrocute** [comment]
@SleeplessOne1917 I think CI is stuck. :(

**Comment by dessalines** [comment]
I moved our CI to a different server, *hopefully* that should be okay now.

**Comment by alectrocute** [comment]
I fixed the merge conflicts introduced by #1043, this PR should be good to merge!

**Comment by alectrocute** [comment]
I have no idea why CI is failing on the merged changes. I'm not seeing any type errors locally. Hmm.

**Comment by SleeplessOne1917** [comment]
@alectrocute I'm not sure why CI isn't working unfortunately. It's complaining about code that doesn't exist anymore.

**Review by SleeplessOne1917 — APPROVED** [review]
LGTM besides the mysterious CI error.

**Pull request merged**